File: backend/SafeSpot.Domain/Entities/Announcement.cs
namespace SafeSpot.Domain.Entities;

public class Announcement : BaseEntity
{
    public long UserId { get; set; }
    public virtual User User { get; set; }

    public long ShelterId { get; set; }
    public virtual Shelter Shelter { get; set; }

    public required string Title { get; set; }
    public required string Text { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/api/models/Announcement.ts
export type AnnouncementDto = {
  id: number;
  title: string;
  text: string;
  imageUrl?: string | null;
};

export type CreateAnnouncementRequest = {
  shelterId: number;
  title: string;
  text: string;
  imageUrl?: string | null;
}

export type UpdateAnnouncementRequest = {
  announcementId: number;
  title: string;
  text: string;
  imageUrl?: string | null;
}